File: Day-4/ArrayPartition.js
function arrayPartitionSum(nums) {
  // Sort the array
  nums.sort((a, b) => a - b);
  
  let sum = 0;
  // Add every second element starting from index 0
  for (let i = 0; i < nums.length; i += 2) {
      sum += nums[i];
  }
  
  return sum;
}

// Test cases here

File: Day-4/ArrayPartition.go
package main

import (
	"fmt"
	"sort"
)

func arrayPartition(nums []int) int {
	// Sort the array
	sort.Ints(nums)

	sum := 0
	// Add every second element starting from index 0
	for i := 0; i < len(nums); i += 2 {
		sum += nums[i]
	}

	return sum
}

func main() {
	// Test cases here
}
